lib/Sema/TypeCheckDecl.cpp

```diff
@@ -868,6 +868,12 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
   if (!currentFile || currentDC->isLocalContext())
     return;
 
+  // Cannot define a type called 'Any'
+  if (current->getFullName() == tc.Context.Id_Any) {
```

Comment:
That is true already when in type positions. My initial implementation did as you suggest and took `Any` as a true keyword, but there were problems with `Any.self` expressions slipping through. I settled on this approach because it allowed Any to be an identifier too; I can have another look at it though if you think that is the right trade off, thanks.